### ISSUE TITLE
SC2155 quirk

### ISSUE THREAD

**Issue by mcandre** [koalaman/shellcheck#3435]
SC2155 interacts strangely with bash family `local` scope variables.

```txt
In docker/windows/bin/wix-tree line 30:
    local dir_basename="$(basename "$dir")"
          ^----------^ SC2155 (warning): Declare and assign separately to avoid masking return values.
```

I wouldn't expect `local` variable scoping to alter return values, though I do want to check with other shell coders to make sure. I've only recently introduced local scoping to my shell scripts.

Perhaps a false positive.

I'm also seeing similar ShellCheck warnings when performing `export`s of command outputs as a single line. I acknowledge that POSIX sh family has some wacky logic concerning how exit codes process.

**Comment by e-kwsm** [comment]
Have you ever tried a code yourself or read manual?

```sh
#!/bin/sh
export foo=$(false)
echo $? # 0
```

[`export` (POSIX)](https://pubs.opengroup.org/onlinepubs/9799919799/utilities/V3_chap02.html#export)

> EXIT STATUS
> 0
> Successful completion.
> &gt;0
> At least one operand could not be processed as requested, such as a name operand that could not be exported or an attempt to modify a readonly variable using a name=word operand, or the -p option was specified and a write error occurred.

[`export` (bash)](https://www.gnu.org/software/bash/manual/html_node/Bourne-Shell-Builtins.html#index-export)

> The return status is zero unless an invalid option is supplied, one of the names is not a valid shell variable name, or `-f` is supplied with a name that is not a shell function. 

[`local` (bash)](https://www.gnu.org/software/bash/manual/html_node/Bash-Builtins.html#index-local)
> The return status is zero unless `local` is used outside a function, an invalid *name* is supplied, or *name* is a readonly variable.

**Comment by mcandre** [comment]
For export, I think I understand.

However, for local, am I not certain of what syntax to use instead of a direct inline.

**Comment by e-kwsm** [comment]
```bash
f() {
local foo
foo=$(some command)
}
```
